core/src/executor/fetch.rs

```diff
@@ -365,3 +419,53 @@ pub async fn fetch_join_columns<'a>(
         .try_collect::<Vec<_>>()
         .await
 }
+
+pub async fn fetch_labels<'a>(
+    storage: &dyn GStore,
+    relation: &'a TableFactor,
+    joins: &'a [Join],
+    projection: &'a [SelectItem],
+) -> Result<Vec<String>> {
```

Comment:
```suggestion
pub async fn fetch_labels (
    storage: &dyn GStore,
    relation: &TableFactor,
    joins: &[Join],
    projection: &[SelectItem],
) -> Result<Vec<String>> {
```

Lifetime sharing here doesn't seem necessary.